sdk/eventhub/event-hubs/test/eventProcessorHost.spec.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import chai from "chai";
+const should = chai.should();
+import chaiAsPromised from "chai-as-promised";
+chai.use(chaiAsPromised);
+import debugModule from "debug";
+const debug = debugModule("azure:event-hubs:partitionPump");
+import {
+  EventPosition,
+  EventHubClient,
+  EventData,
+  EventProcessor,
+  PartitionContext,
+  delay
+} from "../src";
+import { EnvVarKeys, getEnvVars } from "./utils/testUtils";
+const env = getEnvVars();
+
+describe("Event processor Host", function(): void {
+  const service = {
+    connectionString: env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
+    path: env[EnvVarKeys.EVENTHUB_NAME]
+  };
+  const client: EventHubClient = new EventHubClient(service.connectionString, service.path);
+  before("validate environment", async function(): Promise<void> {
+    should.exist(
+      env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
+      "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
+    );
+    should.exist(
+      env[EnvVarKeys.EVENTHUB_NAME],
+      "define EVENTHUB_NAME in your environment before running integration tests."
+    );
+  });
+
+  after("close the connection", async function(): Promise<void> {
+    await client.close();
+  });
+
+  describe("Partition processor", function(): void {
+    const receivedEvents: EventData[] = [];
+    let isinitializeCalled = false;
```

Comment:
I think either these variables should be moved into the actual test, or moved into a `beforeEach` section where they can be reinitialized before each test. If I were to add another test now, these variables would have the state from the previous test (state leakage).

Reply:
For now I just moved these variables into the actual test case.